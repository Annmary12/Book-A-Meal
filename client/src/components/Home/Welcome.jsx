import React, { Component } from 'react';
import Header from '../layout/Header.jsx';
import styles from './welcome.scss';

/**
 * @class Welcome
 */
class Welcome extends Component {
  /**
   * @returns {undefined}
   */
  componentDidMount() {
    const animateHungry = () => {
      this.hungryElements.forEach((letterElm, index) => {
        setTimeout(() => letterElm.classList.toggle(styles.bounceIn), index * 250);
      });
    };
    setInterval(animateHungry, 2500);
  }

  hungryElements = [];

  /**
   * @returns {JSX} - React JSX
   */
  render() {
    return (
      <div className={styles.welcome}>
        <Header />
        <div className={styles.mask}>
          <div className={styles.intro}>
            <span className={styles.text}>
              <b>
                {
                  ('Hungry!?').split('').map((letter, index) => (
                    <i
                      key={letter}
                      ref={elm => this.hungryElements[index] = elm}
                      className={styles.hungry}
                    >
                      {letter}
                    </i>
                  ))
                }
              </b>
              <br />
              Book a meal on the go...
            </span>
          </div>
        </div>
      </div>
    );
  }
}

export default Welcome;
