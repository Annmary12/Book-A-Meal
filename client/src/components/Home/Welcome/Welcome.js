import React, { Component } from 'react';
import NavBar from '../../Layout/NavBar/NavBarContainer.js';
import styles from './welcome.scss';

/**
 * @class Welcome
 */
class Welcome extends Component {
  /**
   * @returns {undefined}
   */
  componentDidMount() {
    this.animationInterval = setInterval(this.animateHungry, 2500);
  }

  /**
   * @returns {undefined}
   */
  componentWillUnmount() {
    clearInterval(this.animationInterval);
  }

  hungryElements = [];

  animateHungry = () => {
    this.hungryElements.forEach((letterElm, index) => {
      setTimeout(() => letterElm.classList.toggle(styles.bounceIn), index * 250);
    });
  };

  /**
   * @returns {JSX} - React JSX
   */
  render() {
    return (
      <div className={styles.welcome}>
        <NavBar home />
        <div className={styles.mask}>
          <div className={styles.intro}>
            <span className={styles.text}>
              <span className={styles.hungry}>
                {
                  ('Hungry!?,').split('').map((letter, index) => (
                    <i
                      key={letter}
                      ref={(elm) => { this.hungryElements[index] = elm; }}
                      className={styles.hungryLetter}
                    >
                      {letter}
                    </i>
                  ))
                }
              </span>
              <br />
              Book a meal on the go...
            </span>
          </div>
        </div>
      </div>
    );
  }
}

export default Welcome;
