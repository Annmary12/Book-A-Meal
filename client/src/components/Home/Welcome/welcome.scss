.welcome{
  background: url('//res.cloudinary.com/otse/image/upload/q_auto:low/v1531698049/crispy-fish_ztgkfr.jpg');
  background-repeat: no-repeat;
  background-size: cover;
	width: 100%;
  height: 514px;

  @media only screen and (max-width: 600px){
    height: 362px;
    background-size: 100% 357px;
    width: 101%;
  }
}
.hungry{
  font-size: 112px;

  @media only screen and (max-width: 600px){
    font-size: 60px;
  }
}
.hungryLetter{
  font-style: normal !important;
  margin: 0px;
  cursor: pointer;
  transition: font-size 1s

}

.hungryLetter.bounceIn{
  font-size: 70px;
  @media only screen and (max-width: 600px){
    font-size: 40px;
  }
}

.intro{
  color: #f8fdff;
  font-weight: 900;
  text-align: center;
  font-family: calibri;
  padding-top: 74px;
  font-size: 51px;
  padding-left: 60px;
  padding-right: 60px;
  cursor: pointer;

  @media only screen and (max-width: 600px){
    padding-top: 121px;
    font-size: 20px;
  }

}

.mask{
  width: 100%;
  height: 456px;
  background-color: #0000007d;
  box-shadow: inset 0 -105px 168px -10px #000000;

  @media only screen and (max-width: 600px){
    height: 359px;
  }
}

.text{
  font-family: 'Dancing Script', cursive;
  font-size: 68px;
  @media only screen and (max-width: 600px){
    font-size: 35px;
  }
}