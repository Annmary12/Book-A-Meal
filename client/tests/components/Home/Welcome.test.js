import React from 'react';
import Welcome from '../../../src/components/Home/Welcome/Welcome';

describe('Welcome component', () => {
  const wrapper = mount(
    <Provider store={store}>
      <MemoryRouter>
        <Welcome />
      </MemoryRouter>
    </Provider>
  );

  const welcomeWrapper = wrapper.find('Welcome');

  beforeAll(() => {
    jest.useFakeTimers();
  });

  it('should match snapshot', () => {
    expect(welcomeWrapper).toMatchSnapshot();
  });

  describe('animateHungry method', () => {
    it('should call setTimeout to toggle classes on this.hungryElements', () => {
      const element = {
        classList: {
          toggle: jest.fn()
        }
      }

      welcomeWrapper.instance().animateHungry();

      const hungryElements = welcomeWrapper.instance().hungryElements;

      expect(setTimeout).toBeCalledWith(expect.any(Function), 250);
      expect(setTimeout).toHaveBeenCalledTimes(hungryElements.length);
    });
  });

  it('should call setInterval on component mount', () => {
    welcomeWrapper.instance().componentDidMount();

    expect(setInterval).toBeCalledWith(welcomeWrapper.instance().animateHungry, 2500);
  });

  it('should call clearInterval when component unmounts', () => {
    const animationInterval = welcomeWrapper.instance().animationInterval;
    welcomeWrapper.instance().componentWillUnmount();

    expect(clearInterval).toBeCalledWith(animationInterval);
  });
});