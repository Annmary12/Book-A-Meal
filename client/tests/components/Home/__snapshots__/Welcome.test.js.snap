// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome component should match snapshot 1`] = `
<Welcome>
  <div
    className="welcome"
  >
    <Connect(NavBar)
      home={true}
    >
      <NavBar
        home={true}
        loggedIn={true}
        logout={[Function]}
        pathname="/"
        user={
          Object {
            "email": "mealbooker@gmail.com",
            "fullName": "Otse Obande",
            "id": "e20ac257-86cc-4a6f-a619-0249a201c475",
            "role": "caterer",
            "username": "otseobande",
          }
        }
      >
        <nav
          className="nav home"
        >
          <div
            className="titleArea"
          >
            <Link
              className="title"
              replace={false}
              to="/"
            >
              <a
                className="title"
                href="/"
                onClick={[Function]}
              >
                <img
                  alt="logo"
                  className="logo"
                  src="test-file-stub"
                  width="30"
                />
                <span
                  className="appName"
                >
                  Book-A-Meal
                </span>
              </a>
            </Link>
            <button
              className="navToggle"
              href="#"
              onClick={[Function]}
            >
              <span>
                â˜°
              </span>
            </button>
          </div>
          <div
            className="navMenus"
          >
            <ul
              className="navList"
            >
              <AuthLinks
                pathname="/"
                role="caterer"
              >
                <CatererLinks
                  pathname="/"
                >
                  <li>
                    <PageLink
                      pathname="/"
                      text="Meals"
                      to="/manage-meals"
                    >
                      <Link
                        className=""
                        replace={false}
                        to="/manage-meals"
                      >
                        <a
                          className=""
                          href="/manage-meals"
                          onClick={[Function]}
                        >
                          Meals
                        </a>
                      </Link>
                    </PageLink>
                  </li>
                  <li>
                    <PageLink
                      pathname="/"
                      text="Menus"
                      to="/manage-menus"
                    >
                      <Link
                        className=""
                        replace={false}
                        to="/manage-menus"
                      >
                        <a
                          className=""
                          href="/manage-menus"
                          onClick={[Function]}
                        >
                          Menus
                        </a>
                      </Link>
                    </PageLink>
                  </li>
                  <li>
                    <PageLink
                      pathname="/"
                      text="Orders"
                      to="/manage-orders"
                    >
                      <Link
                        className=""
                        replace={false}
                        to="/manage-orders"
                      >
                        <a
                          className=""
                          href="/manage-orders"
                          onClick={[Function]}
                        >
                          Orders
                        </a>
                      </Link>
                    </PageLink>
                  </li>
                </CatererLinks>
              </AuthLinks>
              <li>
                <Link
                  onClick={[Function]}
                  replace={false}
                  to="/"
                >
                  <a
                    href="/"
                    onClick={[Function]}
                  >
                    Logout
                  </a>
                </Link>
              </li>
            </ul>
          </div>
        </nav>
      </NavBar>
    </Connect(NavBar)>
    <div
      className="mask"
    >
      <div
        className="intro"
      >
        <span
          className="text"
        >
          <span
            className="hungry"
          >
            <i
              className="hungryLetter"
              key="H"
            >
              H
            </i>
            <i
              className="hungryLetter"
              key="u"
            >
              u
            </i>
            <i
              className="hungryLetter"
              key="n"
            >
              n
            </i>
            <i
              className="hungryLetter"
              key="g"
            >
              g
            </i>
            <i
              className="hungryLetter"
              key="r"
            >
              r
            </i>
            <i
              className="hungryLetter"
              key="y"
            >
              y
            </i>
            <i
              className="hungryLetter"
              key="!"
            >
              !
            </i>
            <i
              className="hungryLetter"
              key="?"
            >
              ?
            </i>
            <i
              className="hungryLetter"
              key=","
            >
              ,
            </i>
          </span>
          <br />
          Book a meal on the go...
        </span>
      </div>
    </div>
  </div>
</Welcome>
`;
